packages/playwright-core/src/server/trace/recorder/snapshotterInjected.ts

```diff
@@ -102,6 +102,43 @@ export function frameSnapshotStreamer(snapshotStreamer: string) {
       this._observer = new MutationObserver(list => this._handleMutations(list));
       const observerConfig = { attributes: true, subtree: true };
       this._observer.observe(document, observerConfig);
+      this._refreshListenersWhenNeeded();
+    }
+
+    private _refreshListenersWhenNeeded() {
+      this._refreshListeners();
+
+      const customEventName = '__playwright_snapshotter_global_listeners_check__';
+
+      let seenEvent = false;
+      const handleCustomEvent = () => seenEvent = true;
+      window.addEventListener(customEventName, handleCustomEvent);
+
+      this._observer = new MutationObserver(entries => {
```

Comment:
`this._observer` is already defined at line 102, we need two different variables :)